File: programmers_42586.java
import java.util.*;

class Solution {
    /*
     * 프로그래머스 42586 기능개발 https://programmers.co.kr/learn/courses/30/lessons/42586?language=java
     * 자료구조: 큐
     * 문제 특징: 
        (1) 각 작업마다 배포까지 필요한 날짜를 구하고,
        (2) 각 작업이 배포될 때 함께 배포될 수 있는 작업의 수 구하기
     */
    public Vector<Integer> solution(int[] progresses, int[] speeds) {
        Vector<Integer> answer = new Vector<>();
        
        Queue<Integer> q = new LinkedList<Integer>();
        //각 작업마다 완료까지 필요한 날짜 계산
        for(int i = 0; i< progresses.length; i++){
            double rest = 100.0 - progresses[i];
            q.offer((int)Math.ceil(rest/speeds[i])); //올림-> 2.7일이 걸리면 적어도 3일 후에 배포가 가능
        }
        //작업이 배포될 때, 함께 배포될 수 있는 작업의 갯수 구하기
        while(!q.isEmpty()){
            int count = 1;
            int top = q.poll();
            //함께 배포될 수 있는 작업의 갯수 구하기
            while(!q.isEmpty() && q.peek()<= top){
                count++;
                q.poll();
            }
            answer.add(count);
        }
        return answer;
    }
}

File: programmers_42586.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var result = [1] //값이 1인 원소가 들어있는 배열 생성
    let calc: (Int,Int) -> Int = { Int(ceil(Double(100 - $0) / Double($1))) }
    var days = calc(progresses[0], speeds[0])
    for i in 1...progresses.count-1{
        let needs = calc(progresses[i], speeds[i])
        if(needs<=days){ //이전 작업이랑 동시 배포가 가능할 때
            result[result.count-1] = result[result.count-1]+1
        }else{ //새로 배포해야할 때
            result.append(1)
            days = needs
        }
    }
    
    return result
}
